G_channels.md

```diff
@@ -1,5 +1,385 @@
-#This Guide is Under Construction
+Channels are a really exciting and powerful part of Phoenix, one that allows us to easily add soft-realtime features to our applications. Channels are based on a simple idea - sending and receiving messages. Senders broadcast messages about topics. Receivers subscribe to topics so that they can receive those messages. Senders and receivers can switch roles on the same topic at any time.
 
-####In the meantime, this link might help.
+Since Elixir is based on message passing, you may wonder why we need this extra mechanism to send and receive messages. With Channels, neither senders nor receivers have to be Elixir processes. They can be anything that we can teach to communicate over a Channel - a JavaScript client, an iOS app, another Phoenix application, our watch. Also, messages broadcast over a Channel may have many receivers. Elixir processes communicate one to one.
 
-- [www.phoenixframework.org on Channels](http://www.phoenixframework.org/docs/channels#channels)
+The word "Channel" is really shorthand for a layered system with a number of components. Let's take a quick look at them now so we can see the big picture a little better.
+
+#### The Moving Parts
+- Channel Routes
+
+These are defined in the `router.ex` file as all other routes are. They match on the topic string and dispatch matching requests to the given Channel module. The star character `*` acts as a wildcard matcher, so in the following example route, requests for `sample_topic:pizza` and `sample_topic:oranges` would both be dispatched to the `SampleTopicChannel`.
+
+```elixir
+socket "/ws", HelloPhoenix do
+  channel "sample_topic:*", SampleTopicChannel
+end
+```
+For more details on channel routes, please see the [Routing Guide](http://www.phoenixframework.org/docs/routing).
+
+- Channels
+
+Channels handle requests, so they are similar to Controllers, but there are two key differences. Channel requests can go both directions - incoming and outgoing. Channel connections also persist beyond a single request/response cycle. Channels are the highest level abstraction for realtime communication components in Phoenix.
+
+Each Channel will implement one or more clauses of each of these four callback functions - `join/3`, `leave/2`, `handle_in/3`, and `handle_out/3`.
+
+- PubSub
+
+The `Phoenix.PubSub` module contains functions which are the nuts and bolts of organizing Channel communication - creating topics, deleting topics, subscribing to topics, unsubscribing from topics, broadcasting messages on a topic, listing subscribers to a topic, checking if a topic is active, and checking if a topic exists.
```

Comment:
The `Phoenix.PubSub` module contains functions which are the nuts and bolts of organizing Channel communication - subscribing to topics, unsubscribing from topics, and broadcasting messages on a topic.